webrtc/RTCPeerConnection-helper.js

```diff
@@ -376,96 +308,25 @@ function listenForSSRCs(t, receiver) {
 // It does the heavy lifting of performing signaling handshake,
 // ICE candidate exchange, and waiting for data channel at two
 // end points to open.
-function createDataChannelPair(
-  pc1=new RTCPeerConnection(),
-  pc2=new RTCPeerConnection(),
-  options={})
-{
-  options = Object.assign({}, {
-    channelLabel: '',
-    channelOptions: undefined,
-    doSignaling: true
-  }, options);
-
-  let channel1Options;
-  let channel2Options = null;
-  if (options.channelOptions instanceof Array) {
-    [channel1Options, channel2Options] = options.channelOptions;
-  } else {
-    channel1Options = options.channelOptions;
-  }
-
-  const channel1 = pc1.createDataChannel(options.channelLabel, channel1Options);
-
-  return new Promise((resolve, reject) => {
-    let channel2;
-    let opened1 = false;
-    let opened2 = false;
-
-    function cleanup() {
-      channel1.removeEventListener('open', onOpen1);
-      channel2.removeEventListener('open', onOpen2);
-      channel1.removeEventListener('error', onError);
-      channel2.removeEventListener('error', onError);
-    }
-
-    function onBothOpened() {
-      cleanup();
-      resolve([channel1, channel2]);
+async function createDataChannelPair(
+  pc1 = new RTCPeerConnection(),
+  pc2 = new RTCPeerConnection()) {
+  const pair = [pc1, pc2].map(pc =>
+      pc.createDataChannel('', {negotiated: true, id: 0}));
```

Comment:
If this is intended to be a drop-in behavior for old code, this is completely bogus. The pair is supposed to consist of the datachannel created at one end, paired with the channel that comes out of the ondatachannel event at the other end. You're creating them on both ends.

Reply:
@alvestrand I wouldn't say it's bogus. It's utilizing [negotiated](https://w3c.github.io/webrtc-pc/#dom-rtcdatachannelinit-negotiated) data channels, which are declarative, and meant to be created like this. See https://jsfiddle.net/jib1/ovbc1uzf/ for a working example.

For the usage I audited at the time of writing, it appeared to be a drop-in. Apologies if I missed something.

Reply:
I think this is a bug in Chrome: https://jsfiddle.net/jib1/0u9jn4c6/30/ in Chrome:
```
closed
```
In Firefox:
```
closed
onclose fired
```